File: 백준/Silver/2607. 비슷한 단어/비슷한 단어.py
import sys
input = sys.stdin.readline

N = int(input())
standard = input().rstrip()
cnt = 0

for _ in range(N - 1):
    check = []
    copy_s = list(standard)
    word = input().rstrip()

    for w in word:
        if w not in copy_s:
            check.append(w)
        else:
            copy_s.remove(w)

    if len(check) <= 1 and len(copy_s) <= 1:
        cnt += 1

print(cnt)

File: 백준/Silver/2607. 비슷한 단어/비슷한 단어.java
import java.io.*;
import java.util.*;

public class Main {
    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));

        int N = Integer.parseInt(br.readLine());
        String s = br.readLine();

        List<Character> listS = new ArrayList<>();
        for (int i = 0; i < s.length(); i++) {
            listS.add(s.charAt(i));
        }

        int cnt = 0;
        for (int n = 0; n < N - 1; n++) {
            String word = br.readLine();

            List<Character> check = new ArrayList<>();
            List<Character> copyS = new ArrayList<>(listS);

            for (int i = 0; i < word.length(); i++) {
                char w = word.charAt(i);
                if (copyS.contains(w)) {
                    copyS.remove((Character) w);
                } else {
                    check.add(w);
                }
            }

            if (copyS.size() <= 1 && check.size() <= 1) cnt++;
        }

        System.out.println(cnt);
    }
}